RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc

```diff
@@ -0,0 +1,810 @@
+
+        alpaka::exec<Acc1D>(
+            queue, workDiv, calibDigis{}, clusterThresholds, digis_d->view(), clusters_d->view(), gains, wordCounter);
+
+#ifdef GPU_DEBUG
+        alpaka::wait(queue);
+        std::cout << "countModules kernel launch with " << blocks << " blocks of " << threadsPerBlockOrElementsPerThread
+                  << " threadsPerBlockOrElementsPerThread\n";
+#endif
+
+        alpaka::exec<Acc1D>(
+            queue, workDiv, countModules<TrackerTraits>{}, digis_d->view(), clusters_d->view(), wordCounter);
+
+        auto moduleStartFirstElement =
+            cms::alpakatools::make_device_view(alpaka::getDev(queue), clusters_d->view().moduleStart(), 1u);
+        alpaka::memcpy(queue, nModules_Clusters_h, moduleStartFirstElement);
+        const auto threadsPerBlockFindClus =
+            512;  //((TrackerTraits::maxPixInModule / 16 + 128 - 1) / 128) * 128;  /// should be larger than maxPixInModule/16 aka (maxPixInModule/maxiter in the kernel)
+        const auto workDivMaxNumModules =
+            cms::alpakatools::make_workdiv<Acc1D>(numberOfModules, threadsPerBlockFindClus);
+        // NB: With present findClus() / chargeCut() algorithm,
+        // threadPerBlock (GPU) or elementsPerThread (CPU) = 256 show optimal performance.
+        // Though, it does not have to be the same number for CPU/GPU cases.
+
+#ifdef GPU_DEBUG
+        std::cout << " findClus kernel launch with " << numberOfModules << " blocks of " << threadsPerBlockFindClus
+                  << " threadsPerBlockOrElementsPerThread\n";
+#endif
+
+        alpaka::exec<Acc1D>(
+            queue, workDivMaxNumModules, findClus<TrackerTraits>{}, digis_d->view(), clusters_d->view(), wordCounter);
+
+#ifdef GPU_DEBUG
+        alpaka::wait(queue);
+#endif
+
+        // apply charge cut
+        alpaka::exec<Acc1D>(queue,
+                            workDivMaxNumModules,
+                            ::pixelClustering::clusterChargeCut<TrackerTraits>{},
+                            digis_d->view(),
+                            clusters_d->view(),
+                            clusterThresholds,
+                            wordCounter);
+        // count the module start indices already here (instead of
+        // rechits) so that the number of clusters/hits can be made
+        // available in the rechit producer without additional points of
+        // synchronization/ExternalWork
+
+        // MUST be ONE block
+        const auto workDivOneBlock = cms::alpakatools::make_workdiv<Acc1D>(1u, 1024u);
+        alpaka::exec<Acc1D>(queue, workDivOneBlock, fillHitsModuleStart<TrackerTraits>{}, clusters_d->view());
+
+        // last element holds the number of all clusters
+        const auto clusModuleStartLastElement = cms::alpakatools::make_device_view(
+            alpaka::getDev(queue),
+            const_cast<uint32_t const *>(clusters_d->view().clusModuleStart() + numberOfModules),
```

Comment:
I'd suggest to remove `const_cast` as not really necessary (given it is a red flag in most cases). The `const`ness of the pointed-to memory region could be achieved with
```suggestion
            clusters_d->const_view().clusModuleStart() + numberOfModules,
```